Update the Transfer Learning for NN Architectures - Multi-Modal Model part (#13)
johko/computer-vision-course: course/2 Neural Network Architectures/2.3 Multi-Modal Models/2.3.3 Transfer Learning/README.md

# Transfer Learning / Fine Tuning for Multimodal Models
In this directory, you can find multiple notebooks that introduce how to fine-tune Multimodal models to downstream task like:
 - Image-to-Text / Text-to-Image
 - Document Classification
 - Visual Question Answering 
 - Speech-to-Text
 - Text-to-Vide

> We have also included sources on state-of-the-art (SOTA) and current developments in multimodal models, providing you with the most up-to-date information in this field..

Reviewer: This sentence is true for the moment, but knowing the field things change rapidly and in half a year it might already be new SOTA models.
I would vouch for making the claim a bit less "up-to-date"

# Transfer Learning / Fine Tuning for Multimodal Models

> In this directory, you can find multiple notebooks that introduce you how to fine-tuning Multimodal models to downstream task like:

Reviewer: ```suggestion
In this directory, you can find multiple notebooks that introduce how to fine-tune Multimodal models to downstream task like:
```

 - Image-to-Text / Text-to-Image
 - Document Classification

>  - Visual Document Question Answering 

Reviewer: Could we make that more general and call it Visual Question Answering? One of the examples in there can of course be Document specific

```suggestion
 - Visual Question Answering 
```

 - Form Understanding, etc.

> We use different type of Multimodal Models ranging from:

Reviewer: ```suggestion
We use different types of Multimodal Models ranging from:
```

 - LayoutLM family
 - DETR
 - DiT, etc.

> This would include also the setup for training, fine-tuning and inferencing at the end. Feel free to use as sources for your project.

Reviewer: Again I would cut the training part, to make it a bit lighter. In the end fine-tuning is a kind of training I guess, so we don't need to mention that explicitly
```suggestion
This includes also the setup for fine-tuning and inferencing at the end. Feel free to use them as sources for your project.
```

> For Multimodal model related to Text and Image, feel free to start with this [paper](https://arxiv.org/abs/2111.08609).

Reviewer: ```suggestion
For Multimodal models related to Text and Image, feel free to start with this [overview paper](https://arxiv.org/abs/2111.08609).
```

> # Model Introduction

Reviewer: I like the idea of the model introduction, but maybe this is better located in the folder of the tasks that these models are tackling (like Document Classification for LayoutLM) or in the corresponding notebook, so the README here does not get too inflated.

## LayoutLM Family

> LayoutLM is a multimodal model that is develop by the [Microsoft Document AI](https://www.microsoft.com/en-us/research/project/document-ai/overview/) team where they most focusing on develop techniques and models to solve the problem of understanding, extracting and analysing complex document such as PDF, Word Document, HTML, XML, etc.

Reviewer: ```suggestion
LayoutLM is a multimodal model that was developed by the [Microsoft Document AI](https://www.microsoft.com/en-us/research/project/document-ai/overview/) team where they are focusing on develop techniques and models to solve the problem of understanding, extracting and analysing complex document such as PDF, Word Document, HTML, XML, etc.
```

> The backbone of the model is Transformers architecture with Text Embedding to handle text input and Image Embedding to handle image input. Depend on the version, the embedding techniques and pre-trained task might be slightly different.

Reviewer: ```suggestion
The backbone of the model is the Transformers architecture, which uses Text Embeddings to handle text input and Image Embedding to handle image input. Depending on the version, the embedding techniques and pre-trained tasks might be slightly different.
```

> The model can fine-tune downstream various tasks:

Reviewer: ```suggestion
The model can be fine-tuned on various downstream tasks:
```

- Sequence Classification (Document Image Classification)
- Token Classification (Similar like Named Entity Recognition)
- Visual Question Answering (Talk with your Document)

![LayoutLMv2 Architecture](https://th.bing.com/th/id/OIP.uS8KaZ-dipmmY57gkEReZgHaGm?pid=ImgDet&w=897&h=800&rs=1)

> For more paper related to LayoutLM family:

Reviewer: ```suggestion
For more papers related to LayoutLM family:
```

 - Document Classification
 - Visual Question Answering 
 - Speech-to-Text

>  - Text-to-Vide

Author: Typo error: Text-to-Video
